src/connectionbenchmark/uploaddatagenerator.cpp

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "uploaddatagenerator.h"
+#include "leakdetector.h"
+#include "logger.h"
+
+#include <QBuffer>
+#include <QRandomGenerator>
+
+constexpr int32_t MAX_BUFFER_SIZE = 4096;
+
+namespace {
+Logger logger(LOG_MAIN, "UploadDataGenerator");
+}
+
+UploadDataGenerator::UploadDataGenerator(const qint64& totalSize)
```

Comment:
No & here. We support 64bits processor, passing a ref does not give anything more than passing the value. Plus reading this i immagine that you want to change the value owned by the caller.